docs/topics/machinepools.md

```diff
@@ -0,0 +1,127 @@
+> Managed Instance Groups, and Azure Virtual Machine Scale Sets.
+  
+> MachinePool is optional and doesn't replace the need for MachineSet/Machine since not every 
+> infrastructure provider will have an abstraction for managing multiple machines (i.e. bare metal). 
+> Users may always opt to choose MachineSet/Machine when they don't see additional value in 
+> MachinePool for their use case.
+
+*Source: [MachinePool API Proposal](https://github.com/kubernetes-sigs/cluster-api/blob/bf51a2502f9007b531f6a9a2c1a4eae1586fb8ca/docs/proposals/20190919-machinepool-api.md)*
+
+## AzureMachinePool
+Cluster API Provider Azure (CAPZ) has experimental support for `MachinePool` though the infrastructure
+type `AzureMachinePool`. An `AzureMachinePool` corresponds to an [Azure Virtual Machine Scale Set](https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/overview),
+which provides the cloud provider specific resource for orchestrating a group of Virtual Machines.
+
+### Using `clusterctl` to deploy
+To deploy a MachinePool / AzureMachinePool via `clusterctl config` there's a flavor for that.
+
+Make sure to set up your Azure environment as described [here](../getting-started.md#setting-up-your-azure-environment).
+
+```shell
+clusterctl config cluster my-cluster --kubernetes-version v1.17.4 --flavor machinepool > my-cluster.yaml
+```
+
+The template used for this flavor is located [here](../../templates/cluster-template-machinepool.yaml).
+
+### Example MachinePool, AzureMachinePool and KubeadmConfig Resources
+Below is an example of the resources needed to create a pool of Virtual Machines orchestrated with
+a Virtual Machine Scale Set.
+```yaml
+---
+apiVersion: exp.cluster.x-k8s.io/v1alpha3
+kind: MachinePool
+metadata:
+  name: capz-mp-0
+spec:
+  clusterName: capz
+  replicas: 2
+  template:
+    spec:
+      bootstrap:
+        configRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+          kind: KubeadmConfig
+          name: capz-mp-0
+      clusterName: capz
+      infrastructureRef:
+        apiVersion: exp.infrastructure.cluster.x-k8s.io/v1alpha3
+        kind: AzureMachinePool
+        name: capz-mp-0
+      version: v1.17.4
+---
+apiVersion: exp.infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AzureMachinePool
+metadata:
+  name: capz-mp-0
+spec:
+  location: westus2
+  template:
+    osDisk:
+      diskSizeGB: 30
+      managedDisk:
+        storageAccountType: Premium_LRS
+      osType: Linux
+    sshPublicKey: ${YOUR_SSH_PUB_KEY}
+    vmSize: Standard_D2s_v3
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+kind: KubeadmConfig
+metadata:
+  name: capz-mp-0
+spec:
+  files:
+  - content: |
+      {
+        "cloud": "AzurePublicCloud",
+        "tenantId": "tenantID",
+        "subscriptionId": "subscriptionID",
+        "aadClientId": "clientID",
+        "aadClientSecret": "secret",
+        "resourceGroup": "capz",
```

Comment:
Could we make it easy for first time users and change the rg, cluster name etc, to what they set in the CAPI book quickstart?

Reply:
These are just there for example. The flavor actually has environment vars which match the Quickstart.

Reply:
Okay, sounds good!